File: 1855-maximum-distance-between-a-pair-of-values/1855-maximum-distance-between-a-pair-of-values.cpp
class Solution {
public:
    static int maxDistance(vector<int>& nums1, vector<int>& nums2) {
        const int n1=nums1.size(), n2=nums2.size();
        int dist=0;
        for(int i=0, j=0; i<n1 && j<n2; ){
            if (i<=j && nums1[i]<=nums2[j]) dist=max(dist, j-i), j++;
            else if (i<=j) i++;
            else j++;
        }
        return dist;
    }
};

File: 1855-maximum-distance-between-a-pair-of-values/1855-maximum-distance-between-a-pair-of-values.py
class Solution:
    def maxDistance(self, nums1: list[int], nums2: list[int]) -> int:
        i, j = 0, 0
        max_dist = 0

        while i < len(nums1) and j < len(nums2):
            if nums1[i] <= nums2[j]:
                max_dist = max(max_dist, j - i)
                j += 1
            else:
                i += 1

        return max_dist